MapboxDirections/MBRoute.swift

```diff
@@ -38,7 +38,8 @@ open class Route: DirectionsResult {
         case let geometry as JSONDictionary:
             coordinates = CLLocationCoordinate2D.coordinates(geoJSON: geometry)
         case let geometry as String:
-            coordinates = decodePolyline(geometry, precision: 1e5)!
+            let precision = routeOptions.shapeFormat == .polyline ? 1e5 : 1e6
```

Comment:
Probably better to switch on `routeOptions.shapeFormat`?

Reply:
Yeah, probably. Although the line above, `case let geometry as String:` takes care of the case where it's geojson.

Reply:
We need to plumb this through `MBRouteStep` as well. 🤔